src/Microsoft.AspNetCore.WebSockets/WebSocketOptions.cs

```diff
@@ -27,5 +29,11 @@ public WebSocketOptions()
         /// The default is 4kb.
         /// </summary>
         public int ReceiveBufferSize { get; set; }
+
+        /// <summary>
+        /// Set the Origin header values allowed for WebSocket requests to prevent Cross-Site WebSocket Hijacking.
+        /// By default all Origins are allowed.
+        /// </summary>
+        public IList<string> AllowedOrigins { get; internal set; }
```

Comment:
`internal` isn't even necessary. When you do the lower-casing, just store it in a new local instead.